src/openms/include/OpenMS/QC/Suitability.h

```diff
@@ -0,0 +1,173 @@
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Tom Waschischeck $
+// $Authors: Tom Waschischeck $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+
+#include <cfloat>
+#include <map>
+#include <vector>
+
+namespace OpenMS
+{
+  class PeptideIdentification;
+  class PeptideHit;
+
+  /**
+  * @brief This class serves as the library representation @ref TOPP_DatabaseSuitability
+  *
+  * This class holds the functionality of calculating the database suitability.
+  * This can only be done if a combined deNovo+database identification search was performed.
+  * Currently only Comet search is supported.
+  *
+  * Allows for multiple usage, because results are stored internally and can be returned
+  * using getResults().
+  *
+  */
+  class OPENMS_DLLAPI Suitability:
+    public DefaultParamHandler
+  {
+  public:
+
+    /// struct to store results
+    struct SuitabilityData
+    {
+      Size num_top_novo = 0;
+      Size num_top_db = 0;
+      Size num_re_ranked = 0;
+      Size num_interest = 0;
+      double cut_off = DBL_MAX;
+      double suitability = 0;
+    };
+
+    /// Constructor
+    /// Settings are initialized with their default values:
+    /// no_re_rank = false, novo_fract = 1, FDR = 0.01
+    Suitability();
+
+    /// Destructor
+    ~Suitability() = default;
+
+    /**
+    * @brief Computes suitability of a database used to search a mzML
+    *
+    * Counts top deNovo and top database hits. The ratio of db hits vs
+    * all hits yields the suitability.
+    * To re-rank cases, where a de novo peptide scores just higher than
+    * the database peptide, a decoy cut-off is calculated. This functionality
+    * can be turned off. This will result in an underestimated suitability,
+    * but it can solve problems like different search engines or to few decoy hits.
+    * 
+    * Parameters can be set using the functionality of DefaultParamHandler.
+    * Parameters are:
+    *                no_re_rank - re-ranking can be turned off with this
+    *                novo_fract - percent of deNovo peptides to capture in re-ranking
+    *                FDR        - q-value that should be filtered for
+    *
+    * Result is appended to the result member. This allows for multiple usage.
+    *
+    * Attention: This function will calculate q-values for all peptide hits. Your input
```

Comment:
dangerous.
Commonly, one would pass by copy and document that this function needs to change the input.
Advanced users can then `std::move()` the argument into the function to avoid the copy (but loose their handle on the data).